### PR TITLE
docs(guides): add using-webpack-dev-middleware section

### PR DESCRIPTION
**Pull request by navarroaxel**

### CODE REVIEW

File: content/guides/development.md [hunk 1]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
 
 ### Using webpack-dev-middleware
 
-?> Familiar with `webpack-dev-middleware`? We need your help! Please submit a PR to fill in the missing instructions and example here. Make sure to keep it simple as this guide is intended for beginners.
+The `webpack-dev-middleware` provides a way to serve your assets from your express server. The first step is install the webpack and his middleware:
```
Comment (TheDutchCoder) [inline content/guides/development.md]: I would change this to something like:

> `webpack-dev-middleware` provides a way to serve your assets from a server and it's highly configurable as well. Let's install it, as well as an [Express](https://expressjs.com/) server and see how we can use it for development:

File: content/guides/development.md [hunk 2]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
 
+``` bash
+npm install --save-dev webpack webpack-dev-middleware
```
Comment (TheDutchCoder) [inline content/guides/development.md]: No need to install webpack, that's already done, but we do need `express` (if we are to follow your example).

File: content/guides/development.md [hunk 3]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
+```
+
+You need to add a webpack middleware to your express app:
```
Comment (TheDutchCoder) [inline content/guides/development.md]: I would recommend changing this to something like:

> Let's configure `webpack-dev-middleware` to work with our Express server:

File: content/guides/development.md [hunk 4]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
+
+__server.js__
+
+```diff
```
Comment (TheDutchCoder) [inline content/guides/development.md]: If `server.js` is a new file (it is in this case), we don't use a diff, but just a js file.

We also need you to add in a change to the project structure:

__project__

``` diff
  webpack-demo
  |- package.json
  |- webpack.config.js
  |- /dist
  |- /src
    |- index.js
    |- print.js
+   |- server.js
  |- /node_modules
```

File: content/guides/development.md [hunk 5]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
+// ...all your middlewares here.
+
++ app.use(webpackMiddleware(compiler, {
++   publicPath: config.output.publicPath,
++ }));
+
+app.list(3000);
+```
+
+You webpack's config should add some output attributes:
```
Comment (TheDutchCoder) [inline content/guides/development.md]: S/You webpack's / Your webpack

I would also rephrase:

> We need to add some additional properties to the webpack config to tell `webpack-dev-middleware` which directories to use for serving files.

File: content/guides/development.md [hunk 6]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
+  module.exports = {
+    entry: {
+      app: [
+        './src/index.js'
+      ]
+    },
+    devtool: 'inline-source-map',
+    output: {
+      filename: '[name].bundle.js',
++      // no real path is required, just pass "/" but it will work with other paths too.
```
Comment (TheDutchCoder) [inline content/guides/development.md]: This comment is a bit ambiguous, consider being a little more clear, or remove it if it's not that relevant.

File: content/guides/development.md [hunk 7]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
++      path: '/',
++      publicPath: '/assets/'
+    }
+  };
+```
+
+You can activate the hot reload if you add the [`webpack-hot-middleware`](https://github.com/glenjamin/webpack-hot-middleware):
```
Comment (TheDutchCoder) [inline content/guides/development.md]: I would recommend moving this down, as a secondary step. We don't want the reader to do too many things at the same time.

File: content/guides/development.md [hunk 8]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
+npm install --save-dev webpack-hot-middleware
+```
+
+__server.js__
+
+```diff
+const express = require('express');
+const webpack = require('webpack');
+const webpackMiddleware = require('webpack-dev-middleware');
++const webpackHotMiddleware = require('webpack-hot-middleware');
```
Comment (TheDutchCoder) [inline content/guides/development.md]: Please take a look at our other guides (e.g. output management) to see how we format diffs for easier reading :)

File: content/guides/development.md [hunk 9]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
++     new webpack.HotModuleReplacementPlugin(),
++     new webpack.NoErrorsPlugin()
+    ],
+    output: {
+      filename: '[name].bundle.js',
+      path: path.join(__dirname, 'public/assets'),
+      publicPath: '/assets/'
+    }
+  };
+```
```
Comment (TheDutchCoder) [inline content/guides/development.md]: I'm missing instructions on how to actual run the `webpack-dev-middleware`.

I'm assuming the reader would need to add a script to their `package.json` for example?
It's also a good idea to add instructions on which files to open in their browser to see the result(s).

File: content/guides/development.md [hunk 10]
```diff
@@ -237,8 +238,105 @@ T> Now that your server is working, you might want to give [Hot Module Replaceme
...
+
+app.use(webpackMiddleware(compiler, {
+  publicPath: config.output.publicPath,
+}));
+
++app.use(webpackHotMiddleware(compiler));
+
+app.list(3000);
+```
+You need to change you webpack's config to active the hot reload:
```
Comment (TheDutchCoder) [inline content/guides/development.md]: Needs an extra line break here

**Comment by TheDutchCoder** [comment]
@navarroaxel Do you think you'll be working on this still?

I tried installing the stuff myself, but to no avail.

**Comment by diegoprd** [comment]
Guys, i was about to create a PR for this as well... shall we wait a bit longer on @navarroaxel , or shall I create a new PR ?

**Comment by TheDutchCoder** [comment]
@diegoprd I haven't heard back from him in a while, so if you have something ready for a PR feel free to submit it ;)

If I can review it and get it to work locally while following your guide, then I'm a very happy camper! ;)

File: content/guides/development.md (hunk 8 above)
Comment (navarroaxel) [inline content/guides/development.md]: But I use the previous section above as reference.

Comment (TheDutchCoder) [inline content/guides/development.md]: Diffs always have 2 extra spaces before each line to provide space to add the `-` and `+` symbols.

**Comment by TheDutchCoder** [comment]
So I've been messing around with this and I actually seem to have it working.

I'll have to do some cleanup and make sure it flows nicely into our guide, but I think I can add this in.

**Comment by TheDutchCoder** [comment]
Alright I've created a PR for this which tries to explain in more detail how to achieve this: https://github.com/webpack/webpack.js.org/pull/1481

Any additions/comments/help is welcome!

**Pull request closed without merging**